bot/utility/utility.py

```diff
@@ -206,6 +208,50 @@ async def welcome_message(self, user: discord.Member):
                                                               "__gemeinsam__ schaffen wir das!** :muscle:")
         await user.send(content=content, embed=embed)
 
+    @commands.command(name='format')
+    async def format(self, ctx: commands.Context, language: Optional[str], *code: str):
+        """Handler for the `format` command. Formats the provided message using clang-format and prints it using markdown syntax highlighting
+            Refer to the !howto format for an extended explanation of how to use it
+        Args:
+            language: The language abbreviation to highlight the resulting code with
+
+        """
+        try:
+            if language is None:
+                raise Exception("No language provided")
+            code = " ".join(code) if ctx.message.reference is None else ctx.message.reference.resolved.content
+            # mypy would erroneously find an error in the next two lines because it doesn't notice that at this point, `code` is always a str.
+            # You can verify its correctness by renaming the variable `code` as assignment target in the previous and all subsequent lines
+            code = code.removeprefix("```").removeprefix("`").removesuffix("```").removesuffix("`")   # type: ignore # this allows the command to format single- and multiline code blocks
+            result = subprocess.run("clang-format", text=True, input=code.replace("```", "´´´"), check=True, capture_output=True)  # type: ignore
```

Comment:
Does `subprocess` or `clang-format` need additional dependencies from python or on the system? 
If so please add python libs to `requirements.txt` and other prerequisites to the `README`. Also if this limits platform portability (SAM should run on any OS), als state so in the `README`.